flink-java/src/main/java/org/apache/flink/api/java/operators/Keys.java

```diff
@@ -410,6 +410,11 @@ public String toString() {
 			return Arrays.copyOfRange(fields, 0, k+1);
 		}
 	}
+
+	public static <T> boolean checkForAtomicType(TypeInformation<T> typeInfo, String... fields) {
+		return typeInfo.isKeyType() && typeInfo instanceof AtomicType && fields.length == 1 &&
```

Comment:
The checks in this method are correct.
I would structure it a bit different though, to be able to give meaningful error messages.
First check if the type is an AtomicType, if no return false.
Otherwise, do the other checks and throw an `InvalidProgramException` with a meaningful error message if any of them is not true.